FilecoinFoundationWeb/filecoin-foundation
# [UXIT-1358] Send form data to GitHub to open PR

This PR integrates with GitHub API so that user-submitted information about ecosystem projects opens a PR in our repo, like https://github.com/FilecoinFoundationWeb/filecoin-foundation/pull/463 or https://github.com/FilecoinFoundationWeb/filecoin-foundation/pull/462

### Details
The feature primarily consists of backend services that handle GitHub interactions.

These services are called in order when `submitProjectToGithub` is called after a form submission:
1. `getLatestCommitOnMain` - Retrieves the latest commit on the main branch
2. `createBlob` - Creates blobs from content
3. `createTree` - Creates a tree from specified blobs
4. `createCommit` - Creates a  commit associated with a tree
5. `createPR` - Creates a branch and opens a pull request

To authenticate the requests made to GitHub, I created a personal access token that expires on Sun, Aug 25 2024.
This is for testing purposes, we will need to create another one with a longer validity.

### How to test
1. Create a [personal access token](https://github.com/settings/tokens) and add the value to your env under `GITHUB_AUTH_TOKEN`. **The token should have access to the repo scope.**
6. Go to http://localhost:3000/ecosystem-explorer/project-form and fill the form with a project name and an image.
7. Submit the form and wait a few seconds - You should be able to click on `View Submission` to check the PR you just opened.

### Todo before merging

- [x] Create a long-lasting GitHub access token
- [x] Add `GITHUB_AUTH_TOKEN` to Vercel
- [x] Delete `app/ecosystem-explorer/project-form/page.tsx`
